### PR TITLE
[KEYCLOAK-7703] HierarchicalPathBasedKeycloakConfigResolver for more …

### PR DESCRIPTION
**Pull request by grgrzybek**
…fine/coarse grained Keycloak configuration in Karaf

### CODE REVIEW

**Comment by mposolda** [comment]
@grgrzybek Thanks for the PR! Adding Hynek as a reviewer. He will return from PTO on Monday 9th July

File: adapters/oidc/osgi-adapter/src/main/java/org/keycloak/adapters/osgi/PathBasedKeycloakConfigResolver.java [hunk 1]
```diff
@@ -31,10 +34,117 @@
 
 public class PathBasedKeycloakConfigResolver implements KeycloakConfigResolver {
 
+    protected static final Logger log = Logger.getLogger(PaxWebIntegrationService.class);
```
Comment (hmlnarik) [inline adapters/oidc/osgi-adapter/src/main/java/org/keycloak/adapters/osgi/PathBasedKeycloakConfigResolver.java]: Nit: should be `Logger.getLogger(PathBasedKeycloakConfigResolver.class);`

File: adapters/oidc/osgi-adapter/src/main/java/org/keycloak/adapters/osgi/HierarchicalPathBasedKeycloakConfigResolver.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import org.keycloak.adapters.KeycloakDeployment;
+import org.keycloak.adapters.OIDCHttpFacade;
+
+/**
+ * This {@link KeycloakConfigResolver} tries to resolve most specific configuration for given URI path. If not found,
+ * <em>parent</em> path is checked up to top-level path.
+ */
+public class HierarchicalPathBasedKeycloakConfigResolver extends PathBasedKeycloakConfigResolver {
+
+    protected static final Logger log = Logger.getLogger(PaxWebIntegrationService.class);
```
Comment (hmlnarik) [inline adapters/oidc/osgi-adapter/src/main/java/org/keycloak/adapters/osgi/HierarchicalPathBasedKeycloakConfigResolver.java]: Nit: should be `Logger.getLogger(HierarchicalPathBasedKeycloakConfigResolver.class);`

**Comment by hmlnarik** [comment]
@grgrzybek Thanks for the PR! Looks good and certainly deserves documenation. Could you please add a documentation into https://github.com/keycloak/keycloak-documentation/blob/master/securing_apps/topics/oidc/java/fuse7/classic-war.adoc#configuring-the-external-adapter?

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**